torch/csrc/jit/tensorexpr/cuda_codegen.cpp

```diff
@@ -865,6 +865,24 @@ static std::ostream& operator<<(
   return out;
 }
 
+#ifdef USE_ROCM
+static const char* resource_string = R"(
+#include <hip/hip_runtime.h>
+#define POS_INFINITY INFINITY
+#define NEG_INFINITY -INFINITY
+
+template<typename T>
```

Comment:
This is great! maximum looks similar to its Cuda counterpart. Can we resource_string into a common part, and ROCM/Cuda specific part? That gives us more code sharing, and less bug prone down the road.

Reply:
Well, the `__device__` is different. Not sure why CUDA doesn't require it, so there isn't that much to share unless CUDA want's to converge.

Reply:
Does __device__ also work with Cuda? If so, we can make both use __device__. 

On a side note, Cuda compilers sometimes doesn't require __device__ if the function is inlined. But we shouldn't count on that feature.

Reply:
I wanted to leave CUDA alone in this patch as much as I could, but I'll just share the `__device__` version.